source/Octopus.Manager.Tentacle.Tests/WizardFixtures/SetupTentacleWizardFixture.cs

```diff
@@ -69,17 +69,32 @@ public void WhenSettingUpPollingTentacle_ScriptShouldBeCorrectlyGenerated()
                 Environment.NewLine + $"{pathToTentacleExe} service --instance \"{model.InstanceName}\" --install --stop --start";
             script.Should().Be(expectedOutput);
         }
-        
-        static SetupTentacleWizardModel CreateTestSetupTentacleWizardModel()
+
+        [Test]
+        public async Task WhenSettingUpPollingTentacle_TelemetryEventShouldBeSent()
         {
-            var instanceStore = Substitute.For<IApplicationInstanceStore>();
-            var instanceSelectionModel = new InstanceSelectionModel(ApplicationName.Tentacle, instanceStore);
-            const string tentacleInstanceName = "TestInstance";
-            instanceSelectionModel.New(tentacleInstanceName);
+            var telemetryService = new TelemetryServiceBuilder().Build();
+
+            var model = new SetupTentacleWizardModelBuilder()
+                .WithTelemetryService(telemetryService)
+                .Build();
+
+            model.OctopusServerUrl = "http://localhost";
+            model.CommunicationStyle = CommunicationStyle.TentacleActive;
+            model.MachineType = MachineType.DeploymentTarget;
+
+            _ = await model.ReviewAndRunScriptTabViewModel.GenerateAndExecuteScript();
 
-            var model = new SetupTentacleWizardModel(instanceSelectionModel);
-            
-            return model;
+            // TODO: Replace with a less ugly assertion
+            await telemetryService
+                .Received(1)
+                .SendTelemetryEvent(
+                    new Uri("http://localhost"),
+                    Arg.Is<TelemetryEvent>(t =>
+                        t.EventProperties.ContainsKey("Communication Style") &&
+                        t.EventProperties["Communication Style"] == CommunicationStyle.TentacleActive.ToString() &&
+                        t.EventProperties.ContainsKey("Machine Type") &&
+                        t.EventProperties["Machine Type"] == MachineType.DeploymentTarget.ToString()));
```

Comment:
I am aware this assert is ugly as sin 😅 

We can improve this by further refactoring to avoid most (if not all) of the mocks. This is some minor tech debt that I feel is worth accepting in the interest of completing the Tentacle Manager project in a timely fashion, as it will be very easy to modify in the future.